src/Dotnet.Script.Core/Scaffolder.cs

```diff
@@ -100,12 +113,26 @@ private void CreateLaunchConfiguration(string currentWorkingDirectory)
                 Directory.CreateDirectory(vsCodeDirectory);
             }
 
+            // on windows we use this as opportunity to make the file associate for .csx -> dotnet-script
+            // (If dotnet install gives us an install time hook this code should move there)
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                // register dotnet-script as the tool to process .csx files
+                string cmdPath = Path.Combine(Path.GetTempPath(), "register.cmd");
+                using (var textWriter = new StreamWriter(cmdPath))
+                {
+                    textWriter.WriteLine("@reg add HKCU\\Software\\classes\\.csx /f /ve /t REG_SZ -d dotnetscript > nul");
+                    textWriter.WriteLine("@reg add HKCU\\Software\\Classes\\dotnetscript\\Shell\\Open\\Command /f /ve /t REG_EXPAND_SZ /d \"%%USERPROFILE%%\\.dotnet\\tools\\dotnet-script.exe %%1 -- %%*\" > nul");
```

Comment:
dotnet-script can also be installed using Chocolatey or install.ps1/install.sh and then the install location is different.
`ScriptEnvironment` provides the install location. Maybe use that?

Reply:
oh, snap, k I'll do that